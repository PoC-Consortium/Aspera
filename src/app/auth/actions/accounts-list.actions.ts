import { Action } from '@ngrx/store';
import { Account } from '../../lib/model/account';

export enum AccountsListActionTypes {
  SelectAccount = '[Accounts List] Select Account',
}

export class SelectAccount implements Action {
  readonly type = AccountsListActionTypes.SelectAccount;

  constructor(public payload: { account: Account }) {}
}


export type AccountsListActionsUnion = SelectAccount;
