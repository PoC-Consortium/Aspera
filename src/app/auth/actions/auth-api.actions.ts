import { Action } from '@ngrx/store';
import { Account } from '../../lib/model/account';

export enum AuthApiActionTypes {
  LoginSuccess = '[Auth/API] Login Success',
  LoginFailure = '[Auth/API] Login Failure',
  LoginRedirect = '[Auth/API] Login Redirect',
  AccountsListRedirect = '[Auth/API] Accounts List Redirect',
}

export class LoginSuccess implements Action {
  readonly type = AuthApiActionTypes.LoginSuccess;

  constructor(public payload: { account: Account }) {}
}

export class LoginFailure implements Action {
  readonly type = AuthApiActionTypes.LoginFailure;

  constructor(public payload: { error: any }) {}
}

export class LoginRedirect implements Action {
  readonly type = AuthApiActionTypes.LoginRedirect;
}

export class AccountsListRedirect implements Action {
  readonly type = AuthApiActionTypes.AccountsListRedirect;
}

export type AuthApiActionsUnion = LoginSuccess | LoginFailure | LoginRedirect | AccountsListRedirect;
