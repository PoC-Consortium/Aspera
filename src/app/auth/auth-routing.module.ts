import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { LoginPageComponent } from './containers/login-page.component';
import { AccountsListComponent } from './components/accounts-list/accounts-list.component';
import { AccountsListResolver } from './components/accounts-list/accounts-list.resolver';

const routes: Routes = [
  { path: 'login', component: LoginPageComponent },
  { path: 'accounts', component: AccountsListComponent, resolve: { accounts: AccountsListResolver } },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
  providers: [AccountsListResolver]
})
export class AuthRoutingModule {}
