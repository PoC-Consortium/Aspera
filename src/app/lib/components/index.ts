export * from './app-footer';
export * from './app-header';
export * from './control-sidebar';
export * from './menu-aside';
export * from './notification-box';
export * from './send-burst-dialog';
export * from './send-burst-form';
export * from './send-multi-out-form';
