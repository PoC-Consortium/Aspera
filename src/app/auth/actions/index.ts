import * as AuthActions from './auth.actions';
import * as AuthApiActions from './auth-api.actions';
import * as AccountsListActions from './accounts-list.actions';

export { AuthActions, AuthApiActions, AccountsListActions };
