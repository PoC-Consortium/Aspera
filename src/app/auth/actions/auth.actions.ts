import { Action } from '@ngrx/store';
import { Account } from '../../lib/model';

export enum AuthActionTypes {
  Logout = '[Auth] Logout',
  LogoutConfirmation = '[Auth] Logout Confirmation',
  LogoutConfirmationDismiss = '[Auth] Logout Confirmation Dismiss',
}

export class Logout implements Action {
  readonly type = AuthActionTypes.Logout;
  constructor(public payload: { account: Account }) {}
}

export class LogoutConfirmation implements Action {
  readonly type = AuthActionTypes.LogoutConfirmation;
  constructor(public payload: { account: Account }) {}
}

export class LogoutConfirmationDismiss implements Action {
  readonly type = AuthActionTypes.LogoutConfirmationDismiss;
}

export type AuthActionsUnion =
  | Logout
  | LogoutConfirmation
  | LogoutConfirmationDismiss;
