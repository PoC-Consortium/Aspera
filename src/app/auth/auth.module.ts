import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { LoginPageComponent } from './containers/login-page.component';
import { LoginFormComponent } from './components/login-form.component';
import { AccountsListComponent } from './components/accounts-list/accounts-list.component';
import { LogoutConfirmationDialogComponent } from './components/logout-confirmation-dialog.component';

import { AuthEffects } from './effects/auth.effects';
import { reducers } from './reducers';
import { AuthRoutingModule } from './auth-routing.module';
import { MatCardModule, MatFormFieldModule, MatInputModule, MatButtonModule, MatDialogModule, MatIconModule } from '@angular/material';
import { AccountsListEffects } from './effects/accounts-list.effects';

export const COMPONENTS = [
  LoginPageComponent,
  LoginFormComponent,
  LogoutConfirmationDialogComponent,
  AccountsListComponent
];

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    AuthRoutingModule,
    FormsModule,
    MatIconModule,
    MatInputModule,
    MatButtonModule,
    MatFormFieldModule,
    MatCardModule,
    MatFormFieldModule,
    MatInputModule,
    MatButtonModule,
    MatDialogModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects, AccountsListEffects]),
  ],
  declarations: COMPONENTS,
  entryComponents: [LogoutConfirmationDialogComponent],
})
export class AuthModule {}
