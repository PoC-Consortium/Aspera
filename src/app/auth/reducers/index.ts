import {
  createSelector,
  createFeatureSelector,
  ActionReducerMap,
} from '@ngrx/store';
import * as fromRoot from '../../reducers';
import * as fromAuth from './auth.reducer';
import * as fromLoginPage from './login-page.reducer';
import { AuthApiActions } from '../actions';
import { access } from 'fs';

export interface AuthState {
  status: fromAuth.State;
  loginPage: fromLoginPage.State;
}

export interface State extends fromRoot.State {
  auth: AuthState;
}

export const reducers: ActionReducerMap<
  AuthState,
  AuthApiActions.AuthApiActionsUnion
> = {
  status: fromAuth.reducer,
  loginPage: fromLoginPage.reducer,
};

export const selectAuthState = createFeatureSelector<State, AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);
export const getAccount = createSelector(selectAuthStatusState, fromAuth.getAccount);
export const getLoggedIn = createSelector(getAccount, account => !!account);

export const selectLoginPageState = createSelector(
  selectAuthState,
  (state: AuthState) => state.loginPage
);
export const getLoginPageError = createSelector(
  selectLoginPageState,
  fromLoginPage.getError
);
export const getLoginPagePending = createSelector(
  selectLoginPageState,
  fromLoginPage.getPending
);
