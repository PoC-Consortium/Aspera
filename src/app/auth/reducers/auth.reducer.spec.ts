import { reducer } from '../reducers/auth.reducer';
import * as fromAuth from '../reducers/auth.reducer';
import { AuthApiActions, AuthActions } from '../actions/';

import { Account } from '../../lib/model';

describe('AuthReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;

      const result = reducer(undefined, action);

      /**
       * Snapshot tests are a quick way to validate
       * the state produced by a reducer since
       * its plain JavaScript object. These snapshots
       * are used to validate against the current state
       * if the functionality of the reducer ever changes.
       */
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOGIN_SUCCESS', () => {
    it('should add a user set loggedIn to true in auth state', () => {
      const account = new Account();
      const createAction = new AuthApiActions.LoginSuccess({ account });

      const result = reducer(fromAuth.initialState, createAction);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LOGOUT', () => {
    it('should logout a user', () => {
      const initialState = {
        account: new Account(),
      } as fromAuth.State;
      const createAction = new AuthActions.Logout();

      const result = reducer(initialState, createAction);

      expect(result).toMatchSnapshot();
    });
  });
});
