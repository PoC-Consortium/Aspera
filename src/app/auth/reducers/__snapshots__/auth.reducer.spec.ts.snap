// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthReducer LOGIN_SUCCESS should add a user set loggedIn to true in auth state 1`] = `
Object {
  "account": Account {
    "address": undefined,
    "alias": undefined,
    "balance": 0,
    "id": undefined,
    "selected": false,
    "transactions": Array [],
    "type": "offline",
    "unconfirmedBalance": 0,
  },
}
`;

exports[`AuthReducer LOGOUT should logout a user 1`] = `
Object {
  "account": null,
}
`;

exports[`AuthReducer undefined action should return the default state 1`] = `
Object {
  "account": null,
}
`;
